File: backend/src/main/java/com/example/campuspetfoster/entity/HandoverRecord.java
package com.example.campuspetfoster.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("handover_record")
public class HandoverRecord {
    @TableId(type = IdType.AUTO)
    private Long id;
    private Long applicationId;
    private String handoverType;
    private LocalDateTime handoverTime;
    private String fromPerson;
    private String toPerson;
    private String remark;
    private String photoUrl;
}

File: frontend/src/api/handoverRecord.ts
import axios from './axios'

export interface HandoverRecord {
  id: number
  applicationId: number
  handoverType: string
  handoverTime: string
  fromPerson: string
  toPerson: string
  remark: string
  photoUrl: string
}

export interface RecordListParams {
  page?: number
  size?: number
  applicationId?: number
  handoverType?: string
}

export interface RecordListResponse {
  total: number
  list: HandoverRecord[]
}

export const getHandoverRecordList = (params: RecordListParams = {}): Promise<RecordListResponse> => {
  return axios.get('/handover-records', { params })
}

export const getRecordsByApplication = (applicationId: number): Promise<HandoverRecord[]> => {
  return axios.get(`/handover-records/application/${applicationId}`)
}

export const getHandoverRecordById = (id: number): Promise<HandoverRecord> => {
  return axios.get(`/handover-records/${id}`)
}

export const createHandoverRecord = (data: Partial<HandoverRecord>): Promise<HandoverRecord> => {
  return axios.post('/handover-records', data)
}

export const updateHandoverRecord = (id: number, data: Partial<HandoverRecord>): Promise<HandoverRecord> => {
  return axios.put(`/handover-records/${id}`, data)
}

export const deleteHandoverRecord = (id: number): Promise<string> => {
  return axios.delete(`/handover-records/${id}`)
}

export const getHandoverStats = (): Promise<Record<string, number>> => {
  return axios.get('/handover-records/stats/summary')
}
